### PR TITLE
systemtests: added python-fd-plugin-local-fileset-test

### PR DESCRIPTION
**Pull request by astoorangi**
Tests plugin functionality from the fd.

### CODE REVIEW

File: systemtests/CMakeLists.txt [hunk 1]
```diff
@@ -96,7 +96,7 @@ set(logdir "log")
 set(mon_dir_password "mon_dir_password")
 set(mon_fd_password "mon_fd_password")
 set(mon_sd_password "mon_sd_password")
-set(plugindir ${PROJECT_BINARY_DIR}/plugindir)
+set(plugindir ${CMAKE_SOURCE_DIR}/core/src/plugins)
```
Comment (joergsteffens) [inline systemtests/CMakeLists.txt]: Is this change required? Is plugindir used afterwards?
I agree, that this setting is better than the setting before. However, it is still wrong, as ${CMAKE_SOURCE_DIR}/core/src/plugins) do not contain the plugins, but subdirectories for each daemon (dird, stored, filed) containing the plugins.

File: systemtests/tests/python-fd-plugin-local-fileset-test/etc/bareos/bareos-dir.d/director/bareos-dir.conf.in [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+  # remove comment in next line to load dynamic backends from specified directory
+  Backend Directory = @backenddir@
+
+  # remove comment from "Plugin Directory" to load plugins from specified directory.
+  # if "Plugin Names" is defined, only the specified plugins will be loaded,
+  # otherwise all director plugins (*-dir.so) from the "Plugin Directory".
+  #
+  Plugin Directory = "@plugindir@/dird"
+  Plugin Names = "python"
```
Comment (joergsteffens) [inline systemtests/tests/python-fd-plugin-local-fileset-test/etc/bareos/bareos-dir.d/director/bareos-dir.conf.in]: For this test, the 2 lines above are not required, as it is an filed pluging.

File: systemtests/tests/python-fd-plugin-local-fileset-test/testrunner [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+quit
+END_OF_DATA
+
+run_bareos
+check_for_zombie_jobs storage=File
+stop_bareos
+
+check_two_logs
+list="$(find ${BackupDirectory} -type f)"
+check_restore_only_files_diff $list
```
Comment (joergsteffens) [inline systemtests/tests/python-fd-plugin-local-fileset-test/testrunner]: Please add a comment, why you are using check_restore_only_files_diff and not the normal check

File: systemtests/CMakeLists.txt (hunk 1 above)
Comment (astoorangi) [reply systemtests/CMakeLists.txt]: The plugindir variable is currently handled very inconsistently everywhere. An alternative would be to create a plugindir in the test and link all files of the subdirectories (dird, filed etc.) with cmake.

File: systemtests/tests/python-fd-plugin-local-fileset-test/etc/bareos/bareos-dir.d/director/bareos-dir.conf.in (hunk 2 above)
Comment (astoorangi) [reply systemtests/tests/python-fd-plugin-local-fileset-test/etc/bareos/bareos-dir.d/director/bareos-dir.conf.in]: Commented out with new commit

File: systemtests/tests/python-fd-plugin-local-fileset-test/testrunner (hunk 3 above)
Comment (astoorangi) [reply systemtests/tests/python-fd-plugin-local-fileset-test/testrunner]: Added comment with the new commit

File: systemtests/CMakeLists.txt [hunk 4]
```diff
@@ -96,9 +96,12 @@ set(logdir "log")
 set(mon_dir_password "mon_dir_password")
 set(mon_fd_password "mon_fd_password")
 set(mon_sd_password "mon_sd_password")
-set(plugindir ${PROJECT_BINARY_DIR}/plugindir)
+set(python_source_plugin_module ${CMAKE_SOURCE_DIR}/core/src/plugins)
```
Comment (franku) [inline systemtests/CMakeLists.txt]:
```suggestion
set(python_plugin_module_src_dir ${CMAKE_SOURCE_DIR}/core/src/plugins)
```

File: systemtests/CMakeLists.txt [hunk 5]
```diff
@@ -96,9 +96,12 @@ set(logdir "log")
...
 set(plugindirtmp ${PROJECT_BINARY_DIR}/plugindirtmp)
 set(rscripts ${PROJECT_BINARY_DIR}/scripts)
+get_target_property(python_binary_plugin_dir python-dir BINARY_DIR)
```
Comment (franku) [inline systemtests/CMakeLists.txt]:
```suggestion
get_target_property(director_plugin_binary_path python-dir BINARY_DIR)
```
